harness/testTypedArray.js

```diff
@@ -120,3 +121,27 @@ function testTypedArrayConversions(byteConversionValues, fn) {
     });
   });
 }
+
+function createTypedArrayVariations(TA, values) {
+  const rab = new ArrayBuffer(4 * TA.BYTES_PER_ELEMENT, { maxByteLength: 8 * TA.BYTES_PER_ELEMENT });
+
+  let nonresizable = new TA(values);
+  let fixedLength = new TA(rab, 0, values.length);
+  let lengthTracking = new TA(rab, 0);
+  let fixedLengthWithOffset = new TA(rab, 2 * TA.BYTES_PER_ELEMENT, (values.length / 2));
```

Comment:
This will do something unreasonable if 'values' is too short (because of offset + length being incompatible), is it possible to add an assert that it's not? Is it OK if this creates a 0-length TypedArray?

Reply:
Because if the `values` array is only one item long, it will be empty, you mean? We could check and throw an error or not; I'm not ideological about it. Throwing the error might remind the spec implementor to think about all cases. Or is that not what you mean?

Reply:
Discussed offline.